be/src/exprs/map_apply_expr.cpp

```diff
@@ -38,8 +38,9 @@ StatusOr<ColumnPtr> MapApplyExpr::evaluate_checked(ExprContext* context, Chunk*
     std::vector<ColumnPtr> input_columns;
     NullColumnPtr input_null_map = nullptr;
     MapColumn* input_map = nullptr;
+    ColumnPtr input_map_ptr_ref = nullptr; // hold shared_ptr to avoid early deleted.
     // step 1: get input columns from map(key_col, value_col)
-    for (int i = 1; i < _children.size(); ++i) {
+    for (int i = 1; i < _children.size(); ++i) { // currently only 2 children, may be more in the future
         ColumnPtr child_col = EVALUATE_NULL_IF_ERROR(context, _children[i], chunk);
```

Comment:
```suggestion
        auto child_col = context->evaluate(children[i], chunk);
```

Reply:
done, also applied to other related places.